src/BenchmarkDotNet/Jobs/EnvironmentMode.cs

```diff
@@ -96,6 +97,17 @@ public IReadOnlyList<EnvironmentVariable> EnvironmentVariables
             set => PowerPlanModeCharacteristic[this] = value;
         }
 
+        /// <summary>
+        /// Specifies that BenchmarkDotNet's benchmark runner can handle addresses larger than 2 gigabytes.
+        /// <value>false: Benchmark runner uses the default (64-bit: enabled; 32-bit:disabled). This is the default.</value>
+        /// <value>true: Explicitly specify that BenchmarkDotNet's benchmark runner can handle addresses larger than 2 gigabytes.</value>
+        /// </summary>
+        public bool LargeAddressAware
+        {
+            get => LargeAddressAwareCharacteristic[this];
+            set => LargeAddressAwareCharacteristic[this] = value;
```

Comment:
This concept is Windows-specific, we should add a guard here.

```suggestion
            set
            {
                if (!RuntimeInformation.IsWindows())
                {
                    throw new NotSupportedException("LargeAddressAware is a Windows-specific concept, it must not be set on other OSes.");
                }

                LargeAddressAwareCharacteristic[this] = value;
            }
```